mirrord/cli/src/internal_proxy.rs

```diff
@@ -0,0 +1,128 @@
+//! Internal proxy is accepting connection from local layers and forward it to agent
+//! while having 1:1 relationship - each layer connection is another agent connection.
+//! This might be changed later on.
+//! The main advantage of this design is that we remove kube logic from the layer itself,
+//! thus eliminating bugs that happen due to mix of remote env vars in our code
+//! (previously was solved using envguard which wasn't good enough)
+//! The proxy will either directly connect to an existing agent (currently only used for tests),
+//! or let the [`OperatorApi`] handle the connection.
+
+use std::{
+    net::{Ipv4Addr, SocketAddrV4},
+    time::Duration,
+};
+
+use futures::{stream::StreamExt, SinkExt};
+use mirrord_config::LayerConfig;
+use mirrord_kube::api::{kubernetes::KubernetesAPI, wrap_raw_connection, AgentManagment};
+use mirrord_operator::client::OperatorApi;
+use mirrord_protocol::{ClientMessage, DaemonCodec, DaemonMessage};
+use tokio::{
+    net::{TcpListener, TcpStream},
+    select,
+    sync::mpsc::{self},
+    task::JoinSet,
+    time::timeout,
+};
+
+use crate::error::{InternalProxyError, Result};
+
+const FIRST_CONNECTION_TIMEOUT: u64 = 5;
```

Comment:
Warning: missing docs.

Reply:
Maybe document something related to the [first connection](https://github.com/metalbear-co/mirrord/pull/1059/files#diff-53987291a539547efc68f94456805ec79cec10fdb79bcc7f4a660abcbd4ba2fbR447) notice in the agent?